pkg/util/collections/includes_excludes.go

```diff
@@ -101,6 +105,133 @@ func (ie *IncludesExcludes) ShouldInclude(s string) bool {
 	return ie.includes.Len() == 0 || ie.includes.Has("*") || ie.includes.match(s)
 }
 
+// IncludesExcludesInterface is used as polymorphic IncludesExcludes for Global and scope
+// resources Include/Exclude.
+type IncludesExcludesInterface interface {
+	ShouldInclude(s string) bool                                      // Checkout whether the passed in parameter name should be included.
+	ClusterResourceIsExcluded(resourceName string) bool               // Check whether the cluster resource is excluded, or the cluster filters' values are not default.
+	ClusterFiltersHaveDefaultValue() bool                             // Check whether the cluster filters have default value.
+	PartNamespaceResourcesIncluded() bool                             // Check whether not all namespace scope resources are included.
+	ClusterResourceIsNotIncludedAndExcluded(resourceName string) bool // Check whether the cluster resource is not literally included and excluded.
```

Comment:
This should be `ClusterResourceIsNotIncludedOrExcluded` -- if either included or excluded, it returns false. It doesn't require the resource to be included *and* excluded to return false.

Reply:
```
	for gr := range backedUpGroupResources {
		// back up CRD for resource if found. We should only need to do this if we've backed up at least
		// one item for the resource, when CRD is not literally included or excluded. When CRD is
		// excluded, no need to check whether the resource has CRD. When CRD's included, it is
		// already involved.
		if backupRequest.ResourceIncludesExcludes.ClusterResourceIsNotIncludedAndExcluded(gr.String()) {
			kb.backupCRD(log, gr, itemBackupper)
		}
	}
```

Although the comment here says that CRD is not literally included or excluded, I think it should be not found in both the include and the exclude list. If the CRD is not found in the include list, it's still possible it's in the exclude list, if that it's the case, CRD should be checked. If the CRD is not found in the exclude list, it's still possible it's in the include list, then it's already handled.

I think I should change the comments here to make them aligned.